src/event-bus/EventStore.ts

```diff
@@ -0,0 +1,106 @@
+import { Event, PersistedEvent } from '../shared/events';
+import { Logger } from '../shared/logger';
+import { promisify } from '../shared/promisify';
+import { sleep } from '../shared/sleep';
+
+const eventStoreConstructor = require('eventstore');
+
+export default class EventStore {
+
+  private isConnected = false;
+  private eventStore: any;
+
+  constructor(
+    private readonly publisher: (event: Event<any>, cb: (err: any) => void) => void,
+    private readonly eventStoreConfig: any,
+    private readonly logger: Logger,
+    private readonly numRetries = 1,
+    private readonly sleepFor = 300,
+  ) {
+  }
+
+  public async persistEvent(event: Event<any>) {
+    const tryPersistEvent = async () => {
+      try {
+        await this.connect();
+        const stream = await this.eventStore.getLastEventAsStream({
+          aggregateId: String(event.teamId),
+        });
+        stream.addEvent(event);
+        await this.eventStore.commit(stream);
+        return true;
+      } catch (err) {
+        this.logger.error('Unable to post event %s', event.type, err);
+        this.isConnected = false;
+        return false;
+      }
+    };
+    for (let i = 0; i < this.numRetries + 1; i++) {
+      const persisted = await tryPersistEvent();
+      if (persisted === true) {
```

Comment:
Well, it returns `Promise<boolean>`. It's all too easy to forget the crucial `await` and then if `Promise<false>` will pass the check.